personal_graph/natural.py

```diff
@@ -1,156 +0,0 @@
-import os
-import uuid
-import instructor
-from graphviz import Digraph  # type: ignore
-from openai import OpenAI
-from dotenv import load_dotenv
-from personal_graph.models import KnowledgeGraph, Node, Edge
-from personal_graph.database import (
-    add_node,
-    atomic,
-    connect_nodes,
-    vector_search_node,
-    vector_search_edge,
-    all_connected_nodes,
-)
-
-load_dotenv()
-
-if os.getenv("OPEN_API_KEY"):
-    client = instructor.patch(OpenAI(api_key=os.getenv("OPEN_API_KEY")))
-else:
-    client = None
-
-db_url = os.getenv("LIBSQL_URL")
-auth_token = os.getenv("LIBSQL_AUTH_TOKEN")
-
-
-def generate_graph(query: str) -> KnowledgeGraph:
```

Comment:
can we modularise this to have a concept of a `GraphGenerator` ABC with concrete implementations such as `InstructorGraphGenerator` and eventually users pf the lib could implement their own ones (such as using dspy or langraph etc)

```py
class InstructorGraphGenerator(GraphGenerator):
  def __init__(
    function_calling_model: str = "gpt-3.5-turbo", 
    system_prompt: str = "You are a high quality knowledge graph generator based on the user query for the purpose of generating descriptive, informative, detailed and accurate knowledge graphs. You can generate proper nodes and edges as a knowledge graph.", 
    prompt: str = "Help me describe this user query as a detailed knowledge graph with meaningful relationships that should provide some descriptive attributes(attribute is the detailed and proper information about the edge) and informative labels about the nodes and relationship. Try to make most of the relationships between similar nodes.", 
  ):
    ...

  def generate(query) -> KnowledgeGraph:
    ...
```

Reply:
Yeah, I'll try that.